### ISSUE TITLE
Problems with pcpfm to reproduce notebooks and tutorial

### ISSUE THREAD

**Issue by mansoldof** [shuzhao-li-lab/pcpfm_tutorials#3]
Hi,
I'm trying to reproduce the notebooks without success.
I run asari without problems. But I'm not able to run pcpfm.

See the problems with the examples provided:
```

import pandas as pd
import os

metadata_dicts = []
for x in os.listdir("./MT02Dataset"):
  if x.endswith(".mzML"):
    metadata_dicts.append({
        "File Name": x.rstrip(".mzML"),
        "Sample Type": "Unknown",
        "Filepath": os.path.join(os.path.abspath("./MT02Dataset"), x)
    })
    print(x)
metadata_df = pd.DataFrame(metadata_dicts)
metadata_df.to_csv("metadata.csv")

```
response:

```

batch11_MT_20210805_005.mzML
batch12_MT_20210806_005_20210806121327.mzML
batch13_MT_20210807_005.mzML
batch14_MT_20210808_005.mzML
batch14_MT_20210808_087.mzML
batch14_MT_20210808_089.mzML
batch14_MT_20210808_179.mzML
batch14_MT_20210808_181.mzML

```

try:
```
!pcpfm assemble -o . -j pcpfm_tutorial_basic -s ./metadata.csv
```

error:
```

Attempting:  assemble
Error executing: assemble

        This is the first command in any pcpfm analysis. Starting with a 
        sequence file, specified by '-s', an output directory by '-o'
        and a project name specified by '-j', this will create the 
        experiment directory and initialize the experiment.json. 

        Additional arguments include the ability to add a filter on 
        sequence file entries using the '--filter' option and a JSON
        dictionaries.

        <<TODO>>

        Additionally, the --name_field, and --path_field options will 
        allow the user to specify what field name should be used for the
        name and filepath of the acquisitions. Also using --skip_list and 
        a .txt formatted file containing sample_names to ignore, entries 
        can be excluded from an analysis. 
        
        :param params: This is the master configuration file generated by 
                    parsing the command line arguments plus the defaults. 
        
Expecting value: line 1 column 1 (char 0)

```
Any ideas?
thanks

**Comment by jmmitc06** [comment]
My apologies not getting back to you sooner. I will test this shortly to see if I can replicate. 

Although I cannot rule out I made a mistake somewhere, I know others have used the notebook successfully in google colab (accessible through the link at the top of the notebook in some environments at least). If you are using the notebook outside of colab, that may explain the error. Could you provide some details on the environment you are using to run the notebook? (e.g., your laptop in visual studio using python 3.9)

**Comment by mansoldof** [comment]
Hi, no problem.
Yes I'm using a local notebook. 
See the session info. Thank you

session_info        1.0.0
-----
anyio                       NA
arrow                       1.3.0
asttokens                   NA
attr                        24.2.0
attrs                       24.2.0
babel                       2.16.0
certifi                     2024.08.30
charset_normalizer          3.3.2
comm                        0.2.2
cython_runtime              NA
dateutil                    2.9.0.post0
debugpy                     1.8.5
decorator                   5.1.1
exceptiongroup              1.2.2
executing                   2.1.0
fastjsonschema              NA
fqdn                        NA
idna                        3.10
ipykernel                   6.29.5
isoduration                 NA
jedi                        0.19.1
jinja2                      3.1.4
json5                       0.9.25
jsonpointer                 3.0.0
jsonschema                  4.23.0
jsonschema_specifications   NA
jupyter_events              0.10.0
jupyter_server              2.14.2
jupyterlab_server           2.27.3
markupsafe                  2.1.5
nbformat                    5.10.4
overrides                   NA
packaging                   24.1
parso                       0.8.4
platformdirs                4.3.6
prometheus_client           NA
prompt_toolkit              3.0.47
psutil                      6.0.0
pure_eval                   0.2.3
pydev_ipython               NA
pydevconsole                NA
pydevd                      2.9.5
pydevd_file_utils           NA
pydevd_plugins              NA
pydevd_tracing              NA
pygments                    2.18.0
pyteomics                   NA
pythonjsonlogger            NA
referencing                 NA
requests                    2.32.3
rfc3339_validator           0.1.4
rfc3986_validator           0.1.1
rpds                        NA
send2trash                  NA
six                         1.16.0
sniffio                     1.3.1
socks                       1.7.1
stack_data                  0.6.3
tornado                     6.4.1
traitlets                   5.14.3
typing_extensions           NA
uri_template                NA
urllib3                     2.2.3
wcwidth                     0.2.13
webcolors                   24.8.0
websocket                   1.8.0
yaml                        6.0.2
zmq                         26.2.0
zstandard                   0.23.0

-----
IPython             8.27.0
jupyter_client      8.6.3
jupyter_core        5.7.2
jupyterlab          4.2.5
notebook            7.2.2
-----
Python 3.10.0 (default, Sep 18 2024, 13:26:34) [GCC 11.4.0]
Linux-6.8.0-45-generic-x86_64-with-glibc2.35
-----
Session information updated at 2024-10-03

**Comment by jmmitc06** [comment]
Hmm, I just ran the notebook in colab without issue. The problem you are reporting could occur when the dataset is not present on your local machine. The block of code before generating the metadata downloads the dataset.

Would you mind doing an `ls -alh` in the directory you are running the notebook to verify that the dataset is there? If not, test / rerun the previous code block to attempt to get the dataset. It downloads from github so I doubt access is an issue.

**Comment by mansoldof** [comment]
Hi,
no problem at all.

check out the notebook run:
![image](https://github.com/user-attachments/assets/efeb9cf2-280d-492e-b783-81f2da496e9e)


all files are fine, see them in TOPPView:
![image](https://github.com/user-attachments/assets/ebb86738-5167-408d-9705-190d57ad41d7)

**ls -alh**
```

(base) user@r/pcpfm_test$ ls -alh
total 24K
drwxrwxrwx 1 root root  504 out  4 21:04 .
drwxrwxrwx 1 root root 4,0K out  4 20:59 ..
-rwxrwxrwx 1 root root  11K out  4 21:04 2_basic_pcpfm.ipynb
drwxrwxrwx 1 root root  192 out  4 21:02 .ipynb_checkpoints
-rwxrwxrwx 1 root root    0 out  4 21:03 khipu.log
-rwxrwxrwx 1 root root  926 out  4 21:02 metadata.csv
drwxrwxrwx 1 root root 4,0K out  4 20:59 MT02Dataset
(base) user@r/pcpfm_test$ pcpfm assemble -o . -j pcpfm_tutorial_basic -s ./metadata.csv
Attempting:  assemble
Error executing: assemble

        This is the first command in any pcpfm analysis. Starting with a 
        sequence file, specified by '-s', an output directory by '-o'
        and a project name specified by '-j', this will create the 
        experiment directory and initialize the experiment.json. 

        Additional arguments include the ability to add a filter on 
        sequence file entries using the '--filter' option and a JSON
        dictionaries.

        <<TODO>>

        Additionally, the --name_field, and --path_field options will 
        allow the user to specify what field name should be used for the
        name and filepath of the acquisitions. Also using --skip_list and 
        a .txt formatted file containing sample_names to ignore, entries 
        can be excluded from an analysis. 
        
        :param params: This is the master configuration file generated by 
                    parsing the command line arguments plus the defaults. 
        
Expecting value: line 1 column 1 (char 0)

```

**Comment by jmmitc06** [comment]
Thanks for the extra details. I'll take a look at this and get back to you.

Oct 4, 2024 8:10:18 PM Felipe Mansoldo ***@***.***>:

**Comment by jmmitc06** [comment]
Hi Felipe, 

Thanks for the extra information. I found the problem. 

In assemble I have the option to pass a JSON string to filter the acquisitions. I thought I was checking correctly if that filter was provided or not but I was accidentally trying to parse the empty default value when one is not provided. This has been fixed. 

I also updated the error handling so that when an exception occurs in main, you'll still get a traceback. 

The new version is on pypi and with it installed, I no longer have the issue.,
